RecoTracker/MeasurementDet/plugins/MeasurementTrackerEventProducer.cc

```diff
@@ -112,13 +119,29 @@ void MeasurementTrackerEventProducer::produce(edm::Event& iEvent, const edm::Eve
 
   // put into MTE
   // put into event
-  iEvent.put(std::make_unique<MeasurementTrackerEvent>(*measurementTracker,
-                                                       stripData.release(),
-                                                       pixelData.release(),
-                                                       phase2OTData.release(),
-                                                       stripClustersToSkip,
-                                                       pixelClustersToSkip,
-                                                       phase2ClustersToSkip));
+
+  edm::Handle<VectorHitCollectionNew> vectorHitsHandle;
+  if (useVectorHits) {
+    iEvent.getByToken(thePh2OTVectorHitsLabel, vectorHitsHandle);
+    const VectorHitCollectionNew* phase2OTVectorHits = vectorHitsHandle.product();
```

Comment:
to simplify this a bit:
```cpp
  edm::Handle<VectorHitCollectionNew> vectorHitsHandle;
  iEvent.getByToken(thePh2OTVectorHitsLabel, vectorHitsHandle);
  const VectorHitCollectionNew* phase2OTVectorHits = useVectorHits ? vectorHitsHandle.product() : nullptr;
```
then you just need one `put` statement

Reply:
I'd suggest something along
```cpp
const VectorHitCollectionNew* phase2OTVectorHits = useVectorHits ? &iEvent.get(thePh2OTVectorHitsLabel) : nullptr;
```
to avoid `iEvent.get(thePh2OTVectorHitsLabel)` when it is not supposed to be read (in fact one should get an exception in that case).

Reply:
Ah, that's true. I was thinking the `getByToken` could return an invalid handle if the collection wasn't in the event, but then the handle just wouldn't be used, which is okay. But if the consume isn't registered at all, the token won't work.